src/emerge.cpp

```diff
@@ -595,6 +596,15 @@ EmergeAction EmergeThread::getBlockOrStartGen(
 }
 
 
+void EmergeThread::finishLoad(MapBlock* block)
+{
+	MutexAutoLock envlock(m_server->m_env_mutex);
+	
+	if(block)
```

Comment:
space, and test this before locking to reduce the useless locking (which costs).